File: StrykerDG.FarmForge.Client/farmforge_client/lib/models/inventory/product_type.dart
import 'package:farmforge_client/models/inventory/product_category.dart';

class ProductType {
  int productTypeId;
  int productCategoryId;
  ProductCategory productCategory;
  String name;
  String label;
  int reorderLevel;

  ProductType({
    this.productTypeId,
    this.productCategoryId,
    this.productCategory,
    this.name,
    this.label,
    this.reorderLevel
  });

  factory ProductType.fromMap(Map<String, dynamic> data) {
    ProductCategory category = data['ProductCategory'] != null
      ? ProductCategory.fromMap(data['ProductCategory'])
      : null;

    return ProductType(
      productTypeId: data['ProductTypeId'],
      productCategoryId: data['ProductCategoryId'],
      productCategory: category,
      name: data['Name'],
      label: data['Label'],
      reorderLevel: data['ReorderLevel']
    );
  }

  Map<String, dynamic> toMap() {
    Map<String, dynamic> category = this.productCategory?.toMap();

    return {
      'ProductTypeId': this.productTypeId,
      'ProductCategoryId': this.productCategoryId,
      'ProductCategory': category,
      'Name': this.name,
      'Label': this.label,
      'ReorderLevel': this.reorderLevel
    };
  }
}

File: StykerDG.FarmForge.DataModel/Models/ProductType.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace StrykerDG.FarmForge.DataModel.Models
{
    [Table("ProductType")]
    public class ProductType : BaseModel
    {
        [Key]
        public int ProductTypeId { get; set; }
        public int ProductCategoryId { get; set; }
        [ForeignKey("ProductCategoryId")]
        public ProductCategory ProductCategory { get; set; }
        public string Name { get; set; }
        public string Label { get; set; }
        public int? ReorderLevel { get; set; }
    }
}
